src/background/oauth.js

```diff
@@ -179,6 +181,12 @@ export default class OAuth2 {
     let details;
     let textResponse = await response.text();
 
+    if (response.status === this.HTTP_UNAUTHORIZED) {
```

Comment:
You mention it says `unauthorized_client`, I think you might rather want to handle this in https://github.com/kewisch/gdata-provider/blob/80b080490f9f47a7b196fb0efde6a7a888d4a7aa/src/background/session.js#L265

This way we can use the logic to repeat only once and otherwise note the client is outdated, which I think would make sense?

Reply:
Since we don't have an exhaustive list of all possible reasons, one could also make the argument that we want to restart the authentication if in doubt. As in: If we get a 401 and the reason is not "invalid_client", restart the authentication. I wasn't aware of that handler, I'll relocate the code there once you've responded to this comment.

Reply:
Turns out there _is_ an exhaustive list in the OAuth2 spec. Let me know what you think